test/commons/dom/get-element-stack.js

```diff
@@ -0,0 +1,353 @@
+					'position: absolute;<br/>' +
+					'z-index: 3;' +
+					'</div>' +
+					'</div>'
+			);
+			var stack = getElementStack(vNode).map(function(vNode) {
+				return vNode.actualNode.id;
+			});
+			assert.deepEqual(stack, ['1', '4', 'target', '5', '3', '2']);
+		});
+
+		it('should correctly order children of position elements without z-index', function() {
+			var vNode = queryFixture(
+				'<div id="1" style="position:relative;width:40px;height:40px;">' +
+					'<div id="target" style="width:40px;height:40px;"></div>' +
+					'</div>'
+			);
+			var stack = getElementStack(vNode).map(function(vNode) {
+				return vNode.actualNode.id;
+			});
+			assert.deepEqual(stack, ['target', '1', 'fixture']);
+		});
+
+		it('should correctly order children of position elements with z-index', function() {
+			var vNode = queryFixture(
+				'<div id="1" style="position:relative;width:40px;height:40px;z-index:1">' +
+					'<div id="target" style="width:40px;height:40px;"></div>' +
+					'</div>'
+			);
+			var stack = getElementStack(vNode).map(function(vNode) {
+				return vNode.actualNode.id;
+			});
+			assert.deepEqual(stack, ['target', '1', 'fixture']);
+		});
+
+		it('should handle modals on top of the stack', function() {
+			var vNode = queryFixture(
+				'<main id="1">' +
+					'<div id="2">' +
+					'<p id="target">Hello World</p>' +
+					'</div>' +
+					'</main>' +
+					'<div id="3" style="position:absolute;top:0;left:0;right:0;height:100px"></div>'
+			);
+			var stack = getElementStack(vNode).map(function(vNode) {
+				return vNode.actualNode.id;
+			});
+			assert.deepEqual(stack, ['3', 'target', '2', '1', 'fixture']);
+		});
+
+		it('should handle "pointer-events:none"', function() {
+			var vNode = queryFixture(
+				'<main id="1">' +
+					'<div id="2">' +
+					'<p id="target">Hello World</p>' +
+					'</div>' +
+					'</main>' +
+					'<div id="3" style="position:absolute;top:0;left:0;right:0;height:100px;pointer-events:none"></div>'
+			);
+			var stack = getElementStack(vNode).map(function(vNode) {
+				return vNode.actualNode.id;
+			});
+			assert.deepEqual(stack, ['3', 'target', '2', '1', 'fixture']);
+		});
+
+		it('should return elements left out by document.elementsFromPoint', function() {
+			var vNode = queryFixture(
+				'<main id="1">' +
+					'<div id="2">' +
+					'<label id="3">Foo<input id="target"></label>' +
+					'</div>' +
+					'</main>'
+			);
+			var stack = getElementStack(vNode).map(function(vNode) {
+				return vNode.actualNode.id;
+			});
+			assert.deepEqual(stack, ['target', '3', '2', '1', 'fixture']);
+		});
+
+		it('should not return elements that do not fully cover the target', function() {
+			var vNode = queryFixture(
+				'<div id="1" style="position:relative;">' +
+					'<div id="2" style="position:absolute;width:300px;height:20px;"></div>' +
+					'<p id="target" style="position:relative;z-index:1;height:40px;">Text oh heyyyy <a href="#" id="target">and here\'s <br>a link</a></p>' +
```

Comment:
You should probably give this a width too, to ensure that this test won't fail on <600px viewport width.